### PR TITLE
[DEVHAS-377] update push.yaml for all deployment environments

### PR DESCRIPTION
**Pull request by kim-tsao**
### What does this PR do?:

This is a coordinated fix to separate out our stage and prod environments.  We need to update the script so the commit hash gets updated on all overlays.  

Relies on https://github.com/redhat-appstudio/infra-deployments/pull/1890 to be merged before we can merge this PR

### Which issue(s)/story(ies) does this PR fixes:

https://issues.redhat.com/browse/DEVHAS-377

### PR acceptance criteria:

- [ ] Unit/Functional tests

  

- [ ] Documentation 

   

- [ ] Client Impact

### CODE REVIEW

File: .tekton/push.yaml [hunk 1]
```diff
@@ -17,6 +17,7 @@ spec:
     - name: infra-deployment-update-script
       value: |
         sed -i -e 's|\(https://github.com/redhat-appstudio/application-service/.*?ref=\)\(.*\)|\1{{ revision }}|' -e 's/\(newTag: \).*/\1{{ revision }}/' components/has/base/kustomization.yaml
```
Comment (maysunfaisal) [inline .tekton/push.yaml:19]: should this line be removed since the file is no longer there

Comment (maysunfaisal) [reply .tekton/push.yaml:19]: oh nevermind!

File: config/manager/kustomization.yaml [hunk 2]
```diff
@@ -17,6 +17,9 @@ configMapGenerator:
 - envs:
   - devfile_registry.properties
   name: devfile-registry-config
+- envs:
+    - feature_flag.properties
+  name: feature-flag-config
```
Comment (maysunfaisal) [inline config/manager/kustomization.yaml:22]: where will this feature flag be set?

Comment (kim-tsao) [reply config/manager/kustomization.yaml:22]: It'll be set in the [configMapGenerator](https://github.com/redhat-appstudio/infra-deployments/blob/main/components/has/staging/kustomization.yaml#L8) for each environment overlay:

The changes will look something like this:

For Staging:
```
configMapGenerator:
  - literals:
      - GITHUB_ORG="redhat-appstudio-appdata-staging"
    name: github-config
    behavior: replace
  - literals:
      - ENVIRONMENT="staging"
    name: feature-flag-config
    behavior: replace
```

For Prod:
```
configMapGenerator:
  - literals:
    - GITHUB_ORG="redhat-appstudio-appdata"
    name: github-config
    behavior: replace
  - literals:
    - ENVIRONMENT="production"
    name: feature-flag-config
    behavior: replace
```

There will be one for Dev too.

**Review by maysunfaisal — APPROVED** [review]

**Pull request merged**